drop-in/src/main/java/com/adyen/checkout/dropin/ui/action/ActionComponentDialogFragment.kt

```diff
@@ -151,11 +162,21 @@ internal class ActionComponentDialogFragment : DropInBottomSheetDialogFragment()
     }
 
     private fun handleError(componentError: ComponentError) {
-        when (componentError.exception) {
+        when (val exception = componentError.exception) {
             is CancellationException -> {
                 Logger.d(TAG, "Flow was cancelled by user")
                 onBackPressed()
             }
+            is PermissionException -> {
+                val requiredPermission = exception.requiredPermission
+                Logger.e(TAG, exception.message.orEmpty(), exception)
+                AlertDialog.Builder(requireContext())
+                    .setTitle(R.string.checkout_qr_code_rationale_title_storage_permission)
```

Comment:
These strings will be used now no matter what permission is required, for now this is okay since we only need the storage permission but it's good to add a comment to fetch these strings based on the required permission